fv3core/stencils/saturation_adjustment.py

```diff
@@ -971,82 +948,74 @@ def satadjust_part2(
-                            qa = (q_plus - qstar) / (dq + dq)
-                        else:
-                            qa = (q_plus - qstar) / (2.0 * dq * (1.0 - q_cond))
+                    rqi = (TICE - tin) / (TICE - T_WFR)
+                qstar = rqi * iqs1 + (1.0 - rqi) * wqs1
+                # higher than 10 m is considered "land" and will have higher subgrid
+                # variability
+            mindw = min(1.0, abs(hs) / (10.0 * constants.GRAV))
+            dw = (
+                spec.namelist.dw_ocean
+                + (spec.namelist.dw_land - spec.namelist.dw_ocean) * mindw
+            )
+            # "scale - aware" subgrid variability: 100 - km as the base
+            dbl_sqrt_area = dw * (area ** 0.5 / 100.0e3) ** 0.5
+            maxtmp = 0.01 if 0.01 > dbl_sqrt_area else dbl_sqrt_area
+            hvar = min(0.2, maxtmp)
+            # partial cloudiness by pdf:
+            # assuming subgrid linear distribution in horizontal; this is
+            # effectively a smoother for the binary cloud scheme;
+            # qa = 0.5 if qstar == qpz
+            rh = qpz / qstar
+            # icloud_f = 0: bug - fixed
+            # icloud_f = 1: old fvgfs gfdl) mp implementation
+            # icloud_f = 2: binary cloud scheme (0 / 1)
+            if rh > 0.75 and qpz > 1.0e-8:
+                dq = hvar * qpz
+                q_plus = qpz + dq
+                q_minus = qpz - dq
+                if spec.namelist.icloud_f == 2:  # TODO untested
+                    if qpz > qstar:
+                        qa = 1.0
+                    elif (qstar < q_plus) and (q_cond > 1.0e-8):
+                        qa = min(1.0, ((q_plus - qstar) / dq) ** 2)
                     else:
                         qa = 0.0
-                    # impose minimum cloudiness if substantial q_cond exist
-                    if q_cond > 1.0e-8:
-                        qa = max(spec.namelist.cld_min, qa)
-                    qa = min(1, qa)
-        else:
-            qa = 0.0
+                else:
+                    if qstar < q_minus:
+                        qa = 1.0
+                    else:
+                        if qstar < q_plus:
+                            if spec.namelist.icloud_f == 0:
+                                qa = (q_plus - qstar) / (dq + dq)
+                            else:
+                                qa = (q_plus - qstar) / (2.0 * dq * (1.0 - q_cond))
+                        else:
+                            qa = 0.0
+                        # impose minimum cloudiness if substantial q_cond exist
+                        if q_cond > 1.0e-8:
+                            qa = max(spec.namelist.cld_min, qa)
+                        qa = min(1, qa)
+            else:
+                qa = 0.0
+
+        if not hydrostatic:
+            #pkz = moist_cv.compute_pkz_func(delp, delz, pt, cappa)
+            pkz = exp(cappa * log(constants.RDG * dp / delz * pt))
 
```

Comment:
Yes it turns out that calling `moist_cv.compute_pkz_func` results in an error. The main error message says the following...

```
    def visit_Assign(self, node: ast.Assign):
>       if isinstance(node.value, ast.Call) and node.value.func.id not in gtscript.MATH_BUILTINS:
E       AttributeError: 'Attribute' object has no attribute 'id'
```

I can provide a text file with the whole error message dump if needed.

Reply:
Interesting. I think GT4Py may not support calling functions from other modules as an attribute of a module. I will try and fix that, but in the meantime, you *should* be able to fix it by importing the function **from** the module at the top of this file.

Reply:
oh interesting! @gmao-ckung would you mind trying that, using the compute_pkz function, but instead import it like
from fv3core.stencils.moist_cv import compute_pkz_func

then in the stencil call it with compute_pkz_func() instead of moist_cv.compute_pkz_func() ?

Reply:
> you should be able to fix it by importing the function from the module at the top of this file

Importing the function from the module fixed the issue!  The tests now complete without any errors.